boards/arm/nrf21540dk_nrf52840/nrf21540dk_nrf52840.dts

```diff
@@ -231,12 +232,18 @@ fem_spi: &spi3 {
 
 &flash0 {
 
+	/* Default flash partitioning.
+	 *
+	 * Note: A Zephyr build should use partitions belonging
+	 * to the same flash layout.
+	 */
 	partitions {
 		compatible = "fixed-partitions";
 		#address-cells = <1>;
 		#size-cells = <1>;
 
-		boot_partition: partition@0 {
+	/* Flash layout with MCUboot */
```

Comment:
Really not sure if overlapping partitions is a good idea here. Curious what @galak and @pabigot think.

Reply:
I've added a documentation comment - everywhere where the flash partitioning is defined - stating that the user shall always restrict to use partitions belonging to the same flash layout, i.e. not to use overlapping partitions.